components/NavBar/index.tsx

```diff
@@ -0,0 +1,168 @@
+import AppBar from '@mui/material/AppBar'
+import Toolbar from '@mui/material/Toolbar'
+import Typography from '@mui/material/Typography'
+import Link, { LinkProps } from 'next/link'
+import { useRouter } from 'next/router'
+import React, { useState, useEffect, ReactElement, CSSProperties } from 'react'
+import FilledInput from '@mui/material/FilledInput'
+import Button from '@mui/material/Button'
+import Box from '@mui/system/Box'
+import MenuIcon from '@mui/icons-material/Menu'
+import IconButton from '@mui/material/IconButton'
+
+// NavLink adapted from https://github.com/vercel/next.js/blob/canary/examples/active-class-name/components/ActiveLink.tsx
+type NavLinkProps = LinkProps & { children: string | ReactElement }
+
+const navLinkActiveStyle: CSSProperties = {
+  textDecoration: 'underline',
+  textUnderlineOffset: '0.4rem',
+  textDecorationThickness: '0.15rem',
+}
+const navLinkStyle: CSSProperties = {
+  margin: '0 1rem 0 0',
+  textTransform: 'uppercase',
+  textDecoration: 'none',
+  display: 'inline-flex',
+}
+
+const NavLink = ({ children, ...props }: NavLinkProps) => {
+  const { asPath, isReady } = useRouter()
+
+  const [style, setStyle] = useState<CSSProperties>(navLinkStyle)
+
+  useEffect(() => {
+    // Check if the router fields are updated client-side
+    if (isReady) {
+      // Dynamic route will be matched via props.as
+      // Static route will be matched via props.href
+      const linkPathname = new URL(
+        (props.as || props.href) as string,
+        location.href
+      ).pathname
+
+      // Using URL().pathname to get rid of query and hash
+      const activePathname = new URL(asPath, location.href).pathname
```

Comment:
I think it's in case the current link/provided link in href has auto-scroll-to-id feature which has the hashtag and the id in the url (e.g. `/contact#address`). We want to match the two, disregarding the hash and id (just the pathname).

Reply:
As well as getting rid of the query (e.g. ?a=b) parts